app/src/main/java/com/example/caloriecounter/feature/restaurant/ui/categories/CategoriesContract.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.caloriecounter.feature.restaurant.ui.categories
+
+import com.example.caloriecounter.base.BaseEvent
+import com.example.caloriecounter.base.BaseSideEffect
+import com.example.caloriecounter.base.BaseState
+import com.example.caloriecounter.base.DataState
+import com.example.domain.models.Category
+
+class CategoriesContract {
+    sealed class Event : BaseEvent {
+        data class SelectedCategory(val category: String) : Event()
+        object ReloadData : Event()
+    }
+
+    data class State(
+        val categories: DataState<List<Category>>
+    ) : BaseState
+
+    sealed class Effect : BaseSideEffect {
+        sealed class Navigation : Effect() {
+            data class ToCategoryMeals(val category: String) : Navigation()
+        }
```

Comment:
Also question just curious. Why do you call it Effect if it's really more like coordinator to move from one screen to another ?

Reply:
I have taken this article and its structure like base. At first, `Effect` contained not only navigation. `Effects` for other contracts will have inside different objects.

Reply:
Okay. Have you forgot to add the link to the article ?

Reply:
Yes)